arelle/ValidateXbrlCalcs.py

```diff
@@ -694,58 +695,133 @@ def rangeValue(value, decimals=None, truncate=False) -> tuple[decimal.Decimal, d
+        if precisionInt is None:
             return None
-    else:
+        return insignificantDigitsFromPrecision(value, precisionInt, scale)
+    decimalsInt = finiteDecimalsInt(decimals)
+    if decimalsInt is None:
+        return None
+    return insignificantDigitsFromDecimals(value, decimalsInt, scale)
+
+
+def insignificantDigitsFromDecimals(value: TypeXValue, decimals: int, scale: int | None) -> tuple[Decimal, Decimal] | None:
+    """
+    :param value: Fact value
+    :param decimals: Fact decimals value
+    :param scale: Fact scale value
+    :return: Tuple of significant and insignificant digit portions of the given value based on decimals, if possible. Otherwise None.
+    """
+    if isinf(decimals) or isnan(decimals):
         return None
-    if vDecimal.is_normal() and -28 <= decimals <= 28: # prevent exception with excessive quantization digits
-        if decimals > 0:
-            divisor = ONE.scaleb(-decimals) # fractional scaling doesn't produce scientific notation
-        else:  # extra quantize step to prevent scientific notation for decimal number
-            divisor = ONE.scaleb(-decimals).quantize(ONE, decimal.ROUND_HALF_UP) # should never round
-        insignificantDigits = abs(vDecimal) % divisor
-        if insignificantDigits:
-            return (vDecimal // divisor * divisor,  # truncated portion of number
-                    insignificantDigits)   # nsignificant digits portion of number
+    try:
+        valueDecimal = decimal.Decimal(value)
+        if scale:
+            iScale = int(scale)
+            valueDecimal = valueDecimal.scaleb(iScale)
+    except (decimal.InvalidOperation, ValueError):  # would have been a schema error reported earlier
+        return None
+    if not valueDecimal.is_normal():
+        return None
+    if decimals > 0:
+        divisor = ONE.scaleb(-decimals)  # fractional scaling doesn't produce scientific notation
+    else:  # extra quantize step to prevent scientific notation for decimal number
+        divisor = ONE.scaleb(-decimals).quantize(ONE, decimal.ROUND_HALF_UP) # should never round
+    truncated = DECIMALS_CONTEXT.divide_int(valueDecimal, divisor) * divisor
+    remainder = DECIMALS_CONTEXT.remainder(abs(valueDecimal), divisor)
+    if remainder:
+        return (truncated,  # truncated portion of number
+                remainder)   # nsignificant digits portion of number
     return None
 
 
+def insignificantDigitsFromPrecision(value: TypeXValue, precision: int, scale: int | None) -> tuple[Decimal, Decimal] | None:
+    """
+    :param value: Fact value
+    :param precision: Fact precision value
+    :param scale: Fact scale value
+    :return: Tuple of significant and insignificant digit portions of the given value based on precision, if possible. Otherwise None.
+    """
+    if isinf(precision) or precision == 0 or isnan(precision):
+        return None
+    try:
+        valueFloat = float(value)
```

Comment:
Pre-existing, I don't understand why this logic uses `float` rather than `Decimal`, it seems wrong.  The `pow`/`scaleb` are conceptually redundant, and it seems like it's adding this artificial split to the `insignificantDigits` method (`FromPrecision` and `FromDecimals`), which makes it harder to understand.  It seems like the overall algorithm `insignificantDigits` should be:
```
convert to Decimal
scale # for backwards compat
if precision
  validate precision and convert to decimals
else
  validate decimals
compute truncated/remainder
```
If you do this simpler logic, is there some specific testcase that fails?